tabular/src/autogluon/tabular/predictor/predictor.py

```diff
@@ -690,6 +690,13 @@ def fit(self,
                 ag_args_ensemble = {}
             ag_args_ensemble['save_bag_folds'] = kwargs['_save_bag_folds']
 
+        if time_limit is None:
+            mb_mem_usage_train_data = get_approximate_df_mem_usage(train_data, sample_ratio=0.2).sum() / 1e6
+            num_rows_train = len(train_data)
+            if mb_mem_usage_train_data >= 50 or num_rows_train >= 100000:
+                logger.log(20, f'Warning: Training may take a very long time because `time_limit` was not specified and `train_data` is large ({num_rows_train} samples, {round(mb_mem_usage_train_data, 2)} MB).')
+                logger.log(20, f'\tConsider setting `time_limit` to ensure training finishes within an expected duration or experiment with a small portion of `train_data` to identify an ideal `hyperparameters` configuration.')
```

Comment:
might also mention `presets` in addition to  `hyperparameters`

Reply:
added